connector/connect/common/pom.xml

```diff
@@ -47,18 +47,6 @@
             <groupId>com.google.protobuf</groupId>
             <artifactId>protobuf-java</artifactId>
         </dependency>
-        <dependency>
```

Comment:
Both connect-server and connect-client  have their own independent guava dependencies : )

I think we should use connect-common shade guava, connect-client and connect-server shade connect-common to make connect-server and connect-client use the same guava dependency. 

In addition, because connect-common is included in the connect-server shade, the runnable distribution can also exclude connect-common. 

Do you like this current commit?

Reply:
I tested it a s a's description and it worked as scheduled
I tested as https://github.com/apache/spark/pull/43195#issue-1921234067, i 
<img width="1440" alt="image" src="https://github.com/apache/spark/assets/51110188/0176bb24-264e-4e77-997e-d4cafa9b5aa6">